framework/src/binding/store/primitive.rs

```diff
@@ -77,132 +77,138 @@ impl<S: ServiceState> DefaultStoreUint64<S> {
         }
     }
 
-    fn set_(&mut self, val: u64) -> ProtocolResult<()> {
+    fn inner_set(&mut self, val: u64) -> ProtocolResult<()> {
         self.state.borrow_mut().insert(self.key.clone(), val)?;
         Ok(())
     }
 
     // Add val with self
     // And set the result back to self
-    fn add_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_add(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         match val.overflowing_add(sv) {
-            (sum, false) => self.set_(sum),
+            (sum, false) => self.inner_set(sum),
             _ => Err(StoreError::Overflow.into()),
         }
     }
 
     // Self minus val
     // And set the result back to self
-    fn sub_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_sub(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         if sv >= val {
-            self.set_(sv - val)
+            self.inner_set(sv - val)
         } else {
             Err(StoreError::Overflow.into())
         }
     }
 
     // Multiply val with self
     // And set the result back to self
-    fn mul_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_mul(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         match val.overflowing_mul(sv) {
-            (mul, false) => self.set_(mul),
+            (mul, false) => self.inner_set(mul),
             _ => Err(StoreError::Overflow.into()),
```

Comment:
I think this should return true instead of an error

Reply:
fixed